src/i18n.ts

```diff
@@ -0,0 +1,29 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+
+import en from './locales/en';
+import de from './locales/de';
+
+// Translations catalog
+const resources = {
+  en: {
+    translationsNS: en,
+  },
+  de: {
+    translationsNS: de,
+  },
+};
+
+i18n
+  .use(initReactI18next)
+  .use(LanguageDetector)
```

Comment:
Let's simplify the logic in the component and only rely on configuration. If the component is used in an application, the responsibility lies there.

If no locale or an invalid is passed log a warning in the console and fall back to English.

Reply:
I've removed language detection.